spec/models/invitation_spec.rb

```diff
@@ -78,5 +78,21 @@
         end
       end
     end
+
+    # Regression test for AO3-6094
+    context "Sending an invitation to a valid email" do
+      let(:invite) { build(:invitation, invitee_email: "foo@example.com") }
+
+      it "succeeds and 'sent_at' is set" do
+        expect(invite.save).to be_truthy
+        expect(invite.id).not_to be_nil
+
+        # Reload invite to make sure that change to invite.sent_at is actually persisted
+        # on the database (not just on the 'invite' object currently being created).
+        reloaded_invite = Invitation.find_by(id: invite.id)
+        expect(reloaded_invite).not_to be_nil
```

Comment:
`reload` will throw an error if the invite wasn't saved to the database, so you can skip the nil check here.